module/state_synchronization/indexer/in_memory_indexer_test.go

```diff
@@ -0,0 +1,433 @@
+package indexer
+
+import (
+	"crypto/rand"
+	"fmt"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/ledger/common/pathfinder"
+	"github.com/onflow/flow-go/ledger/complete"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/storage/store/inmemory/unsynchronized"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestNewInMemoryIndexer(t *testing.T) {
+	header := unittest.BlockHeaderFixture()
+	block := unittest.BlockWithParentFixture(header)
+	exeResult := unittest.ExecutionResultFixture(unittest.WithBlock(block))
+	indexer := createInMemoryIndexer(exeResult, header)
+
+	assert.NotNil(t, indexer)
+	assert.Equal(t, header.Height, indexer.registers.LatestHeight())
+}
+
+func TestInMemoryIndexer_IndexBlockData(t *testing.T) {
+	t.Run("Index Single Chunk and Single Register", func(t *testing.T) {
+		header := unittest.BlockHeaderFixture()
+		block := unittest.BlockWithParentFixture(header)
+		blockID := block.ID()
+		exeResult := unittest.ExecutionResultFixture(unittest.WithBlock(block))
+		indexer := createInMemoryIndexer(exeResult, header)
+
+		trie := createTestTrieUpdate(t)
```

Comment:
can you also add a check that `trie.Payloads` is not empty to make sure we are testing what we think we are.